fleet/comms.go

```diff
@@ -33,9 +33,10 @@ type AgentCommsService interface {
 	NotifyAgentGroupMembership(a Agent) error
 	// NotifyAgentPolicies RPC Core -> Agent: Notify Agent of all AgentPolicy it should currently run based on group membership
 	NotifyAgentPolicies(a Agent) error
-
 	// NotifyGroupNewAgentPolicy RPC Core -> AgentGroup
 	NotifyGroupNewAgentPolicy(ctx context.Context, ag AgentGroup, policyID string, ownerID string) error
+	// UnsubscribeAgentGroupMembership unsubscribe the agent membership when delete a agent group
```

Comment:
this can change to NotifyGroupRemoval and send a single RPC message to the group channel